sql/src/main/java/io/crate/lucene/LikeQuery.java

```diff
@@ -41,35 +43,55 @@
 
     @Override
     public Query apply(Function input, LuceneQueryBuilder.Context context) {
-        RefAndLiteral refAndLiteral = RefAndLiteral.of(input);
-        if (refAndLiteral == null) {
+        CaseAwareRefAndLiteral csRrefAndLiteral = CaseAwareRefAndLiteral.of(input);
+        if (csRrefAndLiteral == null) {
             return null;
         }
-        return toQuery(refAndLiteral.reference(), refAndLiteral.literal().value(), context);
+        return toQuery(csRrefAndLiteral.reference(),
+                       csRrefAndLiteral.literal().value(),
+                       csRrefAndLiteral.ignoreCase(),
+                       context);
     }
 
-    static Query toQuery(Reference reference, Object value, LuceneQueryBuilder.Context context) {
+    static Query toQuery(Reference reference, Object value, boolean ignoreCase, LuceneQueryBuilder.Context context) {
         DataType dataType = ArrayType.unnest(reference.valueType());
         return like(
             dataType,
             context.getFieldTypeOrNull(reference.column().fqn()),
-            value
+            value,
+            ignoreCase
         );
     }
 
-    public static Query like(DataType dataType, @Nullable MappedFieldType fieldType, Object value) {
+    public static Query like(DataType dataType, @Nullable MappedFieldType fieldType, Object value, boolean ignoreCase) {
         if (fieldType == null) {
             // column doesn't exist on this index -> no match
             return Queries.newMatchNoDocsQuery("column does not exist in this index");
         }
         if (dataType.equals(DataTypes.STRING)) {
-            return new WildcardQuery(new Term(
+            return createCaseAwareQuery(
                 fieldType.name(),
-                convertSqlLikeToLuceneWildcard(BytesRefs.toString(value))));
+                BytesRefs.toString(value),
+                ignoreCase
+            );
         }
         return fieldType.termQuery(value, null);
     }
 
+    private static final int CASE_INSENSITIVE = CrateRegexCapabilities.FLAG_CASE_INSENSITIVE
+                                                | CrateRegexCapabilities.FLAG_UNICODE_CASE;
+
+    private static Query createCaseAwareQuery(String fieldName,
+                                              String text,
+                                              boolean ignoreCase) {
+        java.util.function.Function<String, String> regexTransformer = ignoreCase ?
+            LikeOperator::patternToRegex
+            :
+            LikeQuery::convertSqlLikeToLuceneWildcard;
+        Term term = new Term(fieldName, regexTransformer.apply(text));
+        return ignoreCase ? new CrateRegexQuery(term, CASE_INSENSITIVE) : new WildcardQuery(term);
```

Comment:
I wonder if we could optimize the query cases here further, given that the terms in the index could be lower-cased if analyzed?